docs/design/logrotate.md

```diff
@@ -0,0 +1,72 @@
+# Ceph CSI Log Rotate Design Document
+
+Log Rotate is the ability to rotate the logs by controlling the size of log files.
+With csi it will be includes to all the csi pods which includes csi daemons and provisoners.
+Using logrotate will create a log file at the hostpath and enable the logrotator sidecar container.
+Logrotator sidecar container will make sure that log files are rotated as per the inputs.
+
+So logrotate will be enabled by,
+
+`CephCSIOperatorConfig CRD`:
+```yaml
+spec:
+    provisioner:
+        privileged: true
+```
+
+Logroate configuration, 
+
+`CephCSIOperatorConfig CRD`:
+
+```yaml
+kind: CephCSIOperatorConfig 
+apiVersion: csi.ceph.io/v1alpha1
+….
+spec: 
+    logLevel: 1 
+    driverSpecDefaults: 
+        log:
+            logLevel: 5
+            # one of: hourly, daily, weekly
+            Periodicity: daily
+            # SUFFIX may be 'M' or 'G' 
+            MaxLogSize: 500M 
+            MaxFiles: 5
+            logHostPath: /var/lib/log 
+```
+
+Similar settings will be overrided by `CephCSIDriver CRD`:
+
+```yaml
+kind: CephCSIDriver 
+apiVersion: csi.ceph.io/v1alpha1 
+metadata: 
+name: "<prefix>.<driver_type>.csi.ceph.com" 
+namespace:     <operator-namespace> 
+spec: 
+    logLevel: 1 
+    driverSpecDefaults: 
```

Comment:
How the `CephCSIDriver` is planned to override the defaults, from the current API I see I have to use the same `driverSpecDefaults` 

Or it be overridden under  `plugin` or `provisioner` heads?